pkg/prometheus/promcfg.go

```diff
@@ -1611,16 +1615,16 @@ func generateRunningFilter() yaml.MapSlice {
 	}
 }
 
-func getLimit(user *uint64, enforced *uint64) *uint64 {
+func getLimit(user *uint64, enforced *uint64, isGlobalLimitNotSupported bool) *uint64 {
 	if enforced == nil {
 		return user
 	}
 
-	if user == nil {
+	if isGlobalLimitNotSupported && user == nil {
```

Comment:
I'd find it easier to have the version inline here
```suggestion
	// If the Prometheus version doesn't support global limit 
	if cg.version.LT(semver.MustParse("2.45.0")) && user == nil {
```